pandas/_libs/tslib.pyx

```diff
@@ -481,18 +492,37 @@ cpdef array_to_datetime(
                 iresult[i] = NPY_NAT
 
             elif PyDateTime_Check(val):
+                if isinstance(val, _Timestamp):
+                    item_reso = val._creso
+                else:
+                    item_reso = NPY_DATETIMEUNIT.NPY_FR_us
+                state.update_creso(item_reso)
+                if infer_reso:
+                    creso = state.creso
                 tz_out = state.process_datetime(val, tz_out, utc_convert)
                 iresult[i] = parse_pydatetime(val, &dts, creso=creso)
 
             elif PyDate_Check(val):
+                item_reso = NPY_DATETIMEUNIT.NPY_FR_s
```

Comment:
Sorry if this was asked before but are we definitely providing second precision with dates? I'm assuming it is the highest precision that we can match on

Reply:
Correct.  This matches what we do with `pd.Timestamp(date_obj)`